topic-operator/src/test/java/io/strimzi/operator/topic/v2/BatchingTopicControllerTest.java

```diff
@@ -309,185 +327,49 @@ public void replicasChangeShouldMoveToPendingFromNew(boolean cruiseControlEnable
         Mockito.doReturn(List.of(outputRt)).when(replicasChangeHandler).requestPendingChanges(anyList());
         Mockito.doReturn(List.of()).when(replicasChangeHandler).requestOngoingChanges(anyList());
 
-        var reconcilableTopics = List.of(inputRt);
-        var differentRfResults = new BatchingTopicController.PartitionedByError<>(
-            List.of(new BatchingTopicController.Pair<>(inputRt, Either.ofRight(currentState))), List.of());
-        Map<ReconcilableTopic, Either<TopicOperatorException, Object>> results = new HashMap<>();
-
         // test
-        new BatchingTopicController(config, Map.of("key", "VALUE"), admin, client, metrics, replicasChangeHandler)
-            .maybeCheckReplicasChanges(reconcilableTopics, differentRfResults, results);
+        var results = new BatchingTopicController(config, Map.of("key", "VALUE"), admin, client, metrics, replicasChangeHandler)
+            .checkReplicasChanges(reconcilableTopics, currentStatesOrError);
 
         if (cruiseControlEnabled) {
-            assertThat(results.keySet().stream().findFirst().get(), is(outputRt));
-            assertThat(results.size(), is(1));
+            assertThat(results.ok().count(), is(1L));
+            assertThat(results.ok().findFirst().get().getKey(), is(outputRt));
         } else {
-            assertThat(results.keySet(), empty());
+            assertThat(results.errors().count(), is(1L));
+            assertThat(results.errors().findFirst().get().getValue(), instanceOf(TopicOperatorException.NotSupported.class));
         }
     }
 
-    @ParameterizedTest
-    @ValueSource(booleans = { true, false })
-    public void replicasChangeShouldMoveToOngoingFromPending(boolean cruiseControlEnabled) {
```

Comment:
Stupid question maybe, but why do we no longer need to test this?

Reply:
This is covered by replicasChangeShouldMoveFromNewToPending. It doesn't really matter which state transition, as I'm mocking the resulting status (i.e. replicasChangeHandler behavior). What I'm really testing is the controller logic around the replicasChangeHandler invocation, which leads to results.ok() or results.error(). Should we use a better name for this test?